src/main/java/de/tum/in/www1/artemis/repository/ExamUserRepository.java

```diff
@@ -20,5 +21,22 @@ public interface ExamUserRepository extends JpaRepository<ExamUser, Long> {
             """)
     ExamUser findByExamIdAndUserId(@Param("examId") long examId, @Param("userId") long userId);
 
-    List<ExamUser> findAllByExamId(long examId);
+    List<ExamUser> findAllByExamId(@Param("examId") long examId);
+
+    @Query("""
+            SELECT new de.tum.in.www1.artemis.web.rest.dto.ExamUserAttendanceCheckDTO(examUser.id, examUser.studentImagePath, examUser.user.login,
+            examUser.user.registrationNumber, examUser.signingImagePath,
+            studentExams.started, studentExams.submitted)
+            FROM ExamUser examUser
+            LEFT JOIN examUser.exam exam
+            LEFT JOIN exam.studentExams studentExams ON studentExams.user.id = examUser.user.id
+            WHERE (exam.id = :#{#examId})
+            AND studentExams.started = true
+            AND examUser.signingImagePath IS NULL OR examUser.signingImagePath = ''
+            AND (examUser.didCheckImage = false
+                    OR examUser.didCheckLogin = false
+                    OR examUser.didCheckRegistrationNumber = false
+                    OR examUser.didCheckName = false)
+            """)
+    List<ExamUserAttendanceCheckDTO> findAllExamUsersWhoDidNotSign(@Param("examId") long examId);
```

Comment:
Why a `List`? I think a `Set` makes more sense here, there is no order.

Reply:
What do you mean by order? Is it order-by clause? And i don't understand why `Set` would make more sense?

Reply:
A `Set` is an unordered collection while a `List` has a order. But since you don't define a `ORDER BY` clause, I'd consider the order as not relevate and would use a `Set` here.

Reply:
Changed it to `Set`, :+1: